--- ml/extracted/Denotation.ml	2024-10-07 21:43:53.000000000 -0400
+++ ml/extracted/Denotation.ml	2024-10-07 20:12:35.000000000 -0400
@@ -33,6 +33,7 @@
 open Subevent
 open Util
 open VellvmIntegers
+open Log
 
 type __ = Obj.t
 let __ = let rec f _ = Obj.repr f in Obj.repr f
@@ -1157,6 +1158,7 @@
 
   let denote_instr i varargs =
     let (pt, i0) = i in
+    let () = write_log_entry (Instr (pt, i0)) in
     (match pt with
      | IId id ->
        (match i0 with
@@ -5824,11 +5826,13 @@
       sum) itree **)
 
   let denote_terminator = function
-  | TERM_Ret v ->
-    let (dt, op) = v in
+    | TERM_Ret v ->
+      let (dt, op) = v in
+      let () = Log.write_log_entry (Ret v) in
     bind (Obj.magic coq_Monad_itree) (Obj.magic denote_exp (Some dt) op)
       (fun dv -> ret (Obj.magic coq_Monad_itree) (Coq_inr dv))
   | TERM_Ret_void ->
+    let () = Log.write_log_entry (Ret_void) in
     ret (Obj.magic coq_Monad_itree) (Coq_inr LP.Events.DV.UVALUE_None)
   | TERM_Br (v, br1, br2) ->
     let (dt, op) = v in
@@ -6093,6 +6097,7 @@
 
   let denote_phi bid_from = function
   | (id, p) ->
+    let () = write_log_entry (Phi_node (id, p, bid_from)) in
     let Phi (dt, args) = p in
     (match assoc eq_dec_raw_id bid_from args with
      | Some op ->
@@ -6270,6 +6275,7 @@
         (ret (Obj.magic coq_Monad_itree))
         (combine_lists_varargs df.df_args args)) (fun x ->
       let (bs, vs) = x in
+      write_log_entry (F_args (df.df_prototype.dc_name, (List.map fst bs)));
       bind (Obj.magic coq_Monad_itree)
         (lift_err
           (coq_ReSum_inr (Obj.magic __) (fun _ _ _ x0 x1 _ ->
