--- ./ml/extracted/Denotation.ml	2024-11-01 23:00:31.000000000 -0400
+++ ./ml/extracted/Denotation.ml	2024-11-01 22:56:00.000000000 -0400
@@ -34,6 +34,7 @@
 open Subevent
 open Util
 open VellvmIntegers
+open Log
 
 type __ = Obj.t
 let __ = let rec f _ = Obj.repr f in Obj.repr f
@@ -1159,6 +1160,7 @@
 
   let denote_instr i varargs =
     let (pt, i0) = i in
+    let _ = write_log_entry (Instr (pt, i0)) in
     (match pt with
      | IId id ->
        (match i0 with
@@ -5415,7 +5417,9 @@
       dtyp terminator -> (__ LP.Events.exp_E, (block_id, LP.Events.DV.uvalue)
       sum) itree **)
 
-  let denote_terminator = function
+  let denote_terminator t =
+    let _ = write_log_entry (Term t) in
+    match t with
   | TERM_Ret v ->
     let (dt, op) = v in
     bind (Obj.magic coq_Monad_itree) (Obj.magic denote_exp (Some dt) op)
@@ -5709,7 +5713,8 @@
       local_id * LP.Events.DV.uvalue) itree **)
 
   let denote_phi bid_from = function
-  | (id, p) ->
+    | (id, p) ->
+      let _ = write_log_entry (Phi_node (id, p, bid_from)) in
     let Phi (dt, args) = p in
     (match assoc eq_dec_raw_id bid_from args with
      | Some op ->
@@ -5886,7 +5891,8 @@
                             __ __ (coq_ReSum_id (fun _ _ -> coq_Id_IFun) __)))))))))))
         (ret (Obj.magic coq_Monad_itree))
         (combine_lists_varargs df.df_args args)) (fun x ->
-      let (bs, vs) = x in
+          let (bs, vs) = x in
+          let _ = write_log_entry (F_args (df.df_prototype.dc_name, (List.map fst bs))) in
       bind (Obj.magic coq_Monad_itree)
         (lift_err
           (coq_ReSum_inr (Obj.magic __) (fun _ _ _ x0 x1 _ ->
